### ISSUE TITLE
imxRowGradients() fails when child models depend upon referenced objects in their sibling models

### ISSUE THREAD

**Issue by RMKirkpatrick** [OpenMx/OpenMx#248]
See https://openmx.ssri.psu.edu/comment/8493#comment-8493 .

**Comment by RMKirkpatrick** [comment]
623057c is relevant, but does not completely resolve this issue.

**Comment by RMKirkpatrick** [comment]
Well, actually, 623057c *does* resolve what I thought was the *only* issue.  But, it still doesn't get `imxRowGradients()` working with OP's model in [the referenced forums thread](https://openmx.ssri.psu.edu/comment/8493).  The real problem is that OP's two child models that contain raw data, "MZ" and "DZ", use fully qualified references to their sibling model, "Base".  BUT, `imxRowGradients()` assumes that each data-containing child model is self-contained in that it can be run separately from the other child models.

I am not really sure how `imxRowGradients()` could be changed to accommodate this case.  I'm not sure it would be worth the trouble to do so, either, since the long-term plan is to write a backend implementation of robust SEs.

**Comment by RMKirkpatrick** [comment]
I changed the name of this issue to better reflect what is still unresolved.

**Comment by mhunter1** [comment]
I think the below example outlines a possible strategy.

```
# Make some model with a multigroup structure and outside references
require(umx)
data(twinData) # ?twinData from Australian twins.
selDVs = c("ht")
twinData[, c("ht1", "ht2")] = twinData[, c("ht1", "ht2")] * 10
mzData <- twinData[twinData$zygosity %in% "MZFF", ]
dzData <- twinData[twinData$zygosity %in% "DZFF", ]
m1 = umxACE(selDVs = selDVs, sep = "", dzData = dzData, mzData = mzData)

# Now for the strategy
custom.compute <- mxComputeSequence(list(mxComputeNumericDeriv(checkGradient=FALSE, hessian=FALSE), mxComputeReportDeriv()))

# currModel: name from contributing model of multigroup model
cm <- m1[['MZ']]

# Replace the full data set with just the candidate row
cm2 <- mxModel(cm, mxData(cm$data$observed[1,,drop=FALSE], 'raw'))

# Create a new model from the full multigroup model that drops all the non-candidate models
# but keeps others
names(m1$submodels)
m2 <- mxModel(model=m1, 'DZ', remove=TRUE)
names(m2$submodels)

# Give that reduced model the custom compute plan and a "multigroup" fit function
#  with one group
m2 <- mxModel(m2, cm2, custom.compute, mxFitFunctionMultigroup('MZ'))

# run it and get the gradient
m2r <- mxRun(m2)
m2r$output$gradient

# Alternatively leave the non-candidate models in but remove all of their data.
#  You'll still need to make the one group multigroup fit function I think
```

**Comment by tbates** [comment]
Just to note that this stops imxRobustSE working with twin models of the cont(top, MZ, DZ) pattern.

https://openmx.ssri.psu.edu/comment/reply/4618/8794

**Comment by RMKirkpatrick** [comment]
I believe this issue was resolved with 5f6ffe9?  Can we close it, @tbates ?

**Comment by tbates** [comment]
Hi @RMKirkpatrick not sure if it fixes it - the comments links are broken now, so not sure what the example script would have been. I will have a read about imxRobustSE() and see

**Comment by RMKirkpatrick** [comment]
@tbates I believe this is the linked forums thread: https://openmx.ssri.psu.edu/node/4618

**Comment by RMKirkpatrick** [comment]
I am 99% sure this issue was resolved with 5f6ffe9.  @tbates , feel free to re-open this issue if that's not the case.
